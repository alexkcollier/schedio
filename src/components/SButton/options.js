export {
  colors,
  allShadeOptions,
  colorNames,
} from '@/utils/component-options'

export const sizes = [
  'small',
  'regular',
  'large',
]

export const types = [
  'outlined',
  'text',
]
