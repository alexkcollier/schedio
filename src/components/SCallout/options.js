export const calloutTypes = ['', 'success', 'warning', 'error']
