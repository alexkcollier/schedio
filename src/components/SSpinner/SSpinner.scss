@keyframes spin {
  to {
    transform: rotate(360deg);
  }
}

@mixin spinner-dimensions($size) {
  height: $size;
  width: $size;
}

.spinner {
  @include spinner-dimensions(1.5em);

  animation: spin duration('slowest') linear infinite;
  fill: none;
  stroke: color('night', 'lightest');
  stroke-dasharray: 66;
  stroke-linecap: round;
  stroke-width: 2;

  &--small {
    @include spinner-dimensions(0.75em);
  }

  &--regular {
    @include spinner-dimensions(1.5em);
  }

  &--medium {
    @include spinner-dimensions(2em);
  }

  &--large {
    @include spinner-dimensions(4em);
  }

  &--xl {
    @include spinner-dimensions(8em);

    stroke-width: 1.5;
  }

  &--color-green {
    stroke: color('green');
  }

  &--color-grey-lighter {
    stroke: color('grey', 'lighter');
  }

  &--color-ice {
    stroke: color('ice');
  }

  &--color-spartan_blue {
    stroke: color('spartan_blue');
  }

  &--color-white {
    stroke: color('white');
  }
}
