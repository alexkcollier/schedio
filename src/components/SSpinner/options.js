export const colors = [
  'green',
  'grey-lighter',
  'ice',
  'night-light',
  'blue',
  'white',
];
export const sizes = [
  'small',
  'regular',
  'medium',
  'large',
  'xl',
];
