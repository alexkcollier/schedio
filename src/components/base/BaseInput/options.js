export const allowed = ['email', 'text', 'password', 'url', 'search', ''].sort()
