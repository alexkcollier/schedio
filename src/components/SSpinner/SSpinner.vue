<template>
  <svg
    :class="classList"
    class="spinner"
    role="status"
    viewBox="0 0 32 32"
  >
    <path d="M16 2 a 14 14 0 0 0 0 28 a 14 14 0 0 0 0 -28" />
  </svg>
</template>

<script>
import { colors, sizes } from './options'

const propError = (propName, values) => console.error(`\`${propName}\` must be one of ${values}`)

export default {
  name: 'SSpinner',

  props: {
    color: {
      type: String,
      default: '',
      validator: c => !c || colors.includes(c) || propError('color', colors),
    },

    size: {
      type: String,
      default: '',
      validator: s => !s || sizes.includes(s) || propError('size', sizes),
    },
  },

  computed: {
    classList () {
      return {
        [`spinner--${this.size}`]: this.size,
        [`spinner--color-${this.color}`]: this.color,
      }
    },
  },
}
</script>

<style lang="scss">
.spinner {
  @include spinner-dimensions(1.5em);

  animation: spin duration('slowest') linear infinite;
  fill: none;
  stroke: color('night', 'lightest');
  stroke-dasharray: 66;
  stroke-linecap: round;
  stroke-width: 2;

  &--small {
    @include spinner-dimensions(0.75em);
  }

  &--regular {
    @include spinner-dimensions(1.5em);
  }

  &--medium {
    @include spinner-dimensions(2em);
  }

  &--large {
    @include spinner-dimensions(4em);
  }

  &--xl {
    @include spinner-dimensions(8em);

    stroke-width: 1.5;
  }

  &--color-green {
    stroke: color('green', 'light');
  }

  &--color-grey-lighter {
    stroke: color('grey', 'lighter');
  }

  &--color-ice {
    stroke: color('ice');
  }

  &--color-spartan_blue {
    stroke: color('spartan_blue');
  }

  &--color-white {
    stroke: color('white');
  }
}
</style>
