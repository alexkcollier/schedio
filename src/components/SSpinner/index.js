import SSpinner from './SSpinner.vue';

export { SSpinner };

export default {
  install: Vue => Vue.component(SSpinner.name, SSpinner),
};
