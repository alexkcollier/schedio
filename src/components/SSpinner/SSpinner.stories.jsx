import { SSpinner } from '@/components/SSpinner'
import { colors, sizes } from '@/components/SSpinner/options'
import { SHeading } from '@/components/SHeading'
import PropList from '@/docs/PropList'
import StoryContainer from '@/docs/StoryContainer'
import { select, withKnobs } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/vue'

storiesOf('Components/SSpinner', module)
  .addDecorator(withKnobs)
  .add('SSpinner', () => {
    const props = {
      size: select('size', ['', ...sizes], '', 'Optional Props'),
      color: select('color', ['', ...colors], '', 'Optional Props')
    }

    return {
      render: h => (
        <StoryContainer>
          <SHeading level="1">Spinner</SHeading>
          <p>
            Spinners indicate that an action is occuring, but its progress is difficult, or not
            practical to measure.
          </p>

          <SHeading level="2">Example</SHeading>
          <SSpinner {...{ props }} />

          {SSpinner.props && <PropList component={SSpinner} />}
        </StoryContainer>
      )
    }
  })
