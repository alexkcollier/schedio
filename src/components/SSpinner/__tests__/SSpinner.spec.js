import { shallowMount } from '@vue/test-utils';
import * as options from '@/components/SSpinner/options';
import SSpinner from '@/components/SSpinner/SSpinner.vue';

describe('SSpinner.vue', () => {
  let wrapper;
  const errorSpy = jest.spyOn(global.console, 'error').mockImplementation(() => { });

  beforeEach(() => {
    wrapper = shallowMount(SSpinner, {
      propsData: {
        progress: '50',
      },
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  options.colors.forEach((color) => {
    it(`can be ${color}`, async () => {
      wrapper.setProps({ color });
      await wrapper.vm.$nextTick();
      expect(errorSpy).not.toHaveBeenCalled();
      expect(wrapper.props('color')).toBe(color);
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  it('only allows valid colors', () => {
    shallowMount(SSpinner, {
      propsData: {
        color: 'this is not a color',
      },
    });
    expect(errorSpy).toHaveBeenCalled();
  });

  options.sizes.forEach((size) => {
    it(`can be ${size}`, async () => {
      wrapper.setProps({ size });
      await wrapper.vm.$nextTick();
      expect(errorSpy).not.toHaveBeenCalled();
      expect(wrapper.props('size')).toBe(size);
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  it('only allows valid sizes', () => {
    shallowMount(SSpinner, {
      propsData: {
        size: 'this is not a size',
      },
    });
    expect(errorSpy).toHaveBeenCalled();
  });
});
