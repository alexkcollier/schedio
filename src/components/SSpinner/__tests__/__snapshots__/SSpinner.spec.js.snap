// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SSpinner.vue can be green 1`] = `
<svg role="status" viewBox="0 0 32 32" class="spinner spinner--color-green" progress="50">
  <path d="M16 2 a 14 14 0 0 0 0 28 a 14 14 0 0 0 0 -28"></path>
</svg>
`;

exports[`SSpinner.vue can be grey-lighter 1`] = `
<svg role="status" viewBox="0 0 32 32" class="spinner spinner--color-grey-lighter" progress="50">
  <path d="M16 2 a 14 14 0 0 0 0 28 a 14 14 0 0 0 0 -28"></path>
</svg>
`;

exports[`SSpinner.vue can be ice 1`] = `
<svg role="status" viewBox="0 0 32 32" class="spinner spinner--color-ice" progress="50">
  <path d="M16 2 a 14 14 0 0 0 0 28 a 14 14 0 0 0 0 -28"></path>
</svg>
`;

exports[`SSpinner.vue can be large 1`] = `
<svg role="status" viewBox="0 0 32 32" class="spinner spinner--large" progress="50">
  <path d="M16 2 a 14 14 0 0 0 0 28 a 14 14 0 0 0 0 -28"></path>
</svg>
`;

exports[`SSpinner.vue can be medium 1`] = `
<svg role="status" viewBox="0 0 32 32" class="spinner spinner--medium" progress="50">
  <path d="M16 2 a 14 14 0 0 0 0 28 a 14 14 0 0 0 0 -28"></path>
</svg>
`;

exports[`SSpinner.vue can be night-light 1`] = `
<svg role="status" viewBox="0 0 32 32" class="spinner spinner--color-night-light" progress="50">
  <path d="M16 2 a 14 14 0 0 0 0 28 a 14 14 0 0 0 0 -28"></path>
</svg>
`;

exports[`SSpinner.vue can be regular 1`] = `
<svg role="status" viewBox="0 0 32 32" class="spinner spinner--regular" progress="50">
  <path d="M16 2 a 14 14 0 0 0 0 28 a 14 14 0 0 0 0 -28"></path>
</svg>
`;

exports[`SSpinner.vue can be small 1`] = `
<svg role="status" viewBox="0 0 32 32" class="spinner spinner--small" progress="50">
  <path d="M16 2 a 14 14 0 0 0 0 28 a 14 14 0 0 0 0 -28"></path>
</svg>
`;

exports[`SSpinner.vue can be spartan_blue 1`] = `
<svg role="status" viewBox="0 0 32 32" class="spinner spinner--color-spartan_blue" progress="50">
  <path d="M16 2 a 14 14 0 0 0 0 28 a 14 14 0 0 0 0 -28"></path>
</svg>
`;

exports[`SSpinner.vue can be white 1`] = `
<svg role="status" viewBox="0 0 32 32" class="spinner spinner--color-white" progress="50">
  <path d="M16 2 a 14 14 0 0 0 0 28 a 14 14 0 0 0 0 -28"></path>
</svg>
`;

exports[`SSpinner.vue can be xl 1`] = `
<svg role="status" viewBox="0 0 32 32" class="spinner spinner--xl" progress="50">
  <path d="M16 2 a 14 14 0 0 0 0 28 a 14 14 0 0 0 0 -28"></path>
</svg>
`;

exports[`SSpinner.vue renders correctly 1`] = `
<svg role="status" viewBox="0 0 32 32" class="spinner" progress="50">
  <path d="M16 2 a 14 14 0 0 0 0 28 a 14 14 0 0 0 0 -28"></path>
</svg>
`;
